Utils/Directory_watcher/dr_watcher.h

```diff
@@ -0,0 +1,135 @@
+#ifndef DR_WATCHER_H
+#define DR_WATCHER_H
+
+#include <algorithm>
+#include <sys/inotify.h>
+#include <iostream>
+#include <sys/stat.h>
+#include <cerrno>
+#include <unistd.h>
+#include <fcntl.h>
+#include <cstring>
+
+enum Status {
+  IS_FILE,
+  IS_DIRECTORY
+};
+
+Status check_status(const std::string& pathname) {
+  struct stat path_stat{};
+  if(stat(pathname.c_str(), &path_stat) == 0) {
+    if(S_ISDIR(path_stat.st_mode)) {
+      return IS_DIRECTORY;
+    }else if(S_ISREG(path_stat.st_mode)) {
+      return IS_FILE;
+    }
+  }else {
+    std::cerr << "Pathname " << pathname << "didn't exist: " << std::strerror(errno) << std::endl;
+    exit(EXIT_FAILURE);
```

Comment:
Utility function does not have moral right neither to exit a process in case of error nor to log anything!!! Throw an exception.